lib/ansible/module_utils/facts/hardware/linux.py

```diff
@@ -700,13 +700,20 @@ def get_device_facts(self):
 
             sg_inq = self.module.get_bin_path('sg_inq')
 
+            # we can get NVMe device's serial number from /sys/block/<name>/device/serial
+            serial_path = "/sys/block/%s/device/serial" % (block)
+
             if sg_inq:
                 device = "/dev/%s" % (block)
                 rc, drivedata, err = self.module.run_command([sg_inq, device])
                 if rc == 0:
                     serial = re.search(r"Unit serial number:\s+(\w+)", drivedata)
                     if serial:
                         d['serial'] = serial.group(1)
+            elif os.path.exists(serial_path):
+                with open(serial_path, 'r') as f:
```

Comment:
Prefer `get_file_content(serial_path)` here -- which will also handle the `.strip()` for you.